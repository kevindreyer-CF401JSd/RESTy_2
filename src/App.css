.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  min-height: 10vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.Form {
  min-height: 20vh;
  display: flex;
  flex-direction: column;
}
#apiurl {
  width: 90%;

}
.RESTmethods {
  display: flex;
  flex-direction: row;
  flex-flow: wrap;
  justify-content: center;
}

.results {
  width: 90%;
  min-height: 60vh;
  border: 5px solid grey;
  align-items: left;
  text-align: left;
}

footer {
  background-color: #282c34;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
}