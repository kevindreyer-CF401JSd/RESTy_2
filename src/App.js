import React from 'react';
import './App.css';

const testApi = 'https://pokeapi.co/api/v2/pokemon/'

class App extends React.Component {
  constructor () {
    super()
    this.state = {
      apiurl: '',
      restMethod: 'get',
      results: ''
    }
  }

  // test = async () => {
  //   const raw = await fetch(testApi);
  //   const data = await raw.json();
  //   console.log('raw',raw);
  //   console.log('data',data);
  // }

  makeAPICall = async () => {
    const raw = await fetch(`${this.state.apiurl}`);
    const data = await raw.json();
    this.setState({
      results: [data]
    })
  }

  handleChange = e => {
    this.setState({
      apiurl: e.target.value
    });
  }

  render () {
    // this.test()
    return (
        <div className="App">
          <header className="App-header">
            <h1>RESTy</h1>
          </header>
            <div className="Form">
              <form>
                <input onChange={this.handleChange} type="text" id="apiurl" name="urlpath" placeholder="api url path"></input>
                <div className="RESTmethods">
                  <span>GET</span><span>POST</span><span>PUT</span><span>DELETE</span>
                </div>
              </form>
                  <button onClick={this.makeAPICall}>Go!</button>
                <article className="results">results: 
                  <div>
                    {this.state.results.length
                      ? this.state.results
                      : 'null'
                    }
                  </div>
                  <p>{this.state.results}</p>
                </article>
            </div>
          <footer>
            Codefellows JS 401d34 Lab 28 Kevin Dreyer
          </footer>
        </div>
    );
  }
}

export default App;