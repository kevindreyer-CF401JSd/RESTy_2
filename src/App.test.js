import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('<App />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('is alive at application start', () => {
    const component = shallow(<App />)
    expect(component.find('div')).toExist()
  })
})


