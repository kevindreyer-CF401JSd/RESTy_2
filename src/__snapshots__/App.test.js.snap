// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      RESTy
    </h1>
  </header>
  <div
    className="Form"
  >
    <form>
      <input
        id="apiurl"
        name="urlpath"
        onChange={[Function]}
        placeholder="api url path"
        type="text"
      />
      <div
        className="RESTmethods"
      >
        <span>
          GET
        </span>
        <span>
          POST
        </span>
        <span>
          PUT
        </span>
        <span>
          DELETE
        </span>
        <button
          onClick={[Function]}
        >
          Go!
        </button>
      </div>
      <article
        className="results"
      >
        results:
        <div>
          null
        </div>
      </article>
    </form>
  </div>
  <footer>
    Codefellows JS 401d34 Lab 28 Kevin Dreyer
  </footer>
</div>
`;
